src/ensembl/src/content/app/entity-viewer/gene-view/components/collapsed-exons-image/CollapsedExonsImage.tsx

```diff
@@ -107,53 +96,22 @@ const ExonsImageWithData = (props: ExonsImageWithDataProps) => {
           height={IMAGE_HEIGHT}
         >
           <g>
-            <g className={styles.splicedRNABlock}>
+            <g className={styles.longestProtein}>
               <rect height={IMAGE_HEIGHT} width={props.width} />
             </g>
             <g>
-              {codingExons.map((exon, index) => (
-                <ExonBlock
-                  key={index}
-                  exon={exon}
-                  className={props.className}
-                  scale={scale}
-                />
-              ))}
+              <rect
+                className={styles.protein}
+                y="8"
+                height={PROTEIN_HEIGHT}
+                width={scale(transcript.product?.length)}
```

Comment:
Same as comment above about availability of the product; but in this case, if we imagine that the product didn't exist, the value of `transcript.product?.length` would be `undefined`. We don't want to pass any undefineds as rect attributes, do we?

Reply:
I've changed it here and also in other places where I came across the same issue